src/Commands/core/SecurityUpdateCommands.php

```diff
@@ -0,0 +1,108 @@
+     *   min-version: Suggested version
+     * @default-fields name,version,min-version
+     *
+     * @return \Consolidation\OutputFormatters\StructuredData\RowsOfFields
+     *
+     * @throws \Exception
+     */
+    public function security()
+    {
+        $this->securityUpdates = [];
+        try {
+            $response_body = file_get_contents('https://raw.githubusercontent.com/drupal-composer/drupal-security-advisories/8.x/composer.json');
+        }
+        catch (\Exception $e) {
+            throw new Exception("Unable to fetch drupal-security-advisories information.");
+
+        }
+        $security_advisories_composer_json = json_decode($response_body, TRUE);
+        $composer_root = Drush::bootstrapManager()->getComposerRoot();
+        $composer_lock_file_path = Path::join($composer_root, 'composer.lock');
+        if (!file_exists($composer_lock_file_path)) {
+            throw new \Exception("Cannot find composer.lock file.");
+        }
+        $composer_lock_contents = json_decode(file_get_contents($composer_lock_file_path), TRUE);
+        foreach ($composer_lock_contents['packages'] as $key => $package) {
+            $name = $package['name'];
+            if (!empty($security_advisories_composer_json['conflict'][$name])) {
+                $conflict_constraints = explode(',', $security_advisories_composer_json['conflict'][$name]);
+                foreach ($conflict_constraints as $conflict_constraint) {
+                    // Only parse constraints that follow pattern like "<1.0.0".
+                    if (substr($conflict_constraint, 0, 1) == '<') {
+                        $min_version = substr($conflict_constraint, 1);
+                        if (Comparator::lessThan($package['version'], $min_version)) {
+                            $this->securityUpdates[$name] = [
+                                'name' => $name,
+                                'version' => $package['version'],
+                                'min-version' => $min_version,
+                            ];
+                        }
+                    }
+                    else {
+                        $this->logger()->warning("Could not parse drupal-security-advisories conflicting version constraint $conflict_constraint for package $name.");
+                    }
+                }
+            }
+        }
+        if ($this->securityUpdates) {
+            // @todo Modernize.
+            drush_set_context('DRUSH_EXIT_CODE', 1);
+            $result = new RowsOfFields($this->securityUpdates);
+            return $result;
+        }
+        else {
+            $this->logger()->info("<info>There are no outstanding security updates for Drupal projects.</info>");
+        }
+    }
+
+    /**
+     * Emit suggested composer command for security updates.
+     *
+     * @hook post-command pm:security
+     */
+    public function preExampleHello($result, CommandData $commandData) {
+        if (isset($this->securityUpdates)) {
```

Comment:
Use !empty() or else you get output when a command failed like at https://travis-ci.org/drush-ops/drush/jobs/290712023. I'm re-running that test now so that failure might fix itself.